api/numaresourcesoperator/v1/helper/nodegroup/nodegroup_test.go

```diff
@@ -185,27 +181,127 @@ func TestFindTrees(t *testing.T) {
 				},
 			},
 		},
+		{
```

Comment:
likewise, good to add these tests (actually: must to have these tests :) ) but they probably better fit in `pkg/validation`

Reply:
I'll remove redundant tests for the function in nodegroup.go and cover what's missing in validation_test.go